config/200-clusterrole.yaml

```diff
@@ -18,21 +33,73 @@ metadata:
     samples.knative.dev/release: devel
     samples.knative.dev/controller: "true"
 rules:
+  # Allow creating events associated with resources we are controlling.
   - apiGroups: [""]
-    resources: ["configmaps", "services", "secrets", "events", "pods"]
-    verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
-  - apiGroups: ["apps"]
-    resources: ["deployments", "deployments/finalizers"] # finalizers are needed for the owner reference of the webhook
-    verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
-  - apiGroups: ["coordination.k8s.io"]
-    resources: ["leases"]
-    verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
+    resources: ["events"]
+    verbs: ["create"]
+
+  # Allow the reconciliation of exactly our validating webhooks.
+  # This is needed for us to patch in caBundle information.
   - apiGroups: ["admissionregistration.k8s.io"]
-    resources: ["mutatingwebhookconfigurations", "validatingwebhookconfigurations"]
-    verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
+    resources: ["validatingwebhookconfigurations"]
+    verbs: ["list", "watch"]
+  - apiGroups: ["admissionregistration.k8s.io"]
+    resources: ["validatingwebhookconfigurations"]
+    verbs: ["get", "update"]
+    resourceNames: ["config.webhook.knative-samples.knative.dev", "validation.webhook.knative-samples.knative.dev"]
+
+  # Allow the reconciliation of exactly our mutating webhooks.
+  # This is needed for us to patch in caBundle information.
+  - apiGroups: ["admissionregistration.k8s.io"]
+    resources: ["mutatingwebhookconfigurations"]
+    verbs: ["list", "watch"]
+  - apiGroups: ["admissionregistration.k8s.io"]
+    resources: ["mutatingwebhookconfigurations"]
+    verbs: ["get", "update"]
+    resourceNames: ["defaulting.webhook.knative-samples.knative.dev"]
```

Comment:
I found this trick in Tekton, so thanks to them for inspiring a bunch of this cleanup.